### DISCUSSION TITLE
自动恢复上次播放文件/列表脚本

### DISCUSSION THREAD

**Discussion by sailovetheworld** [hooke007/mpv_PlayKit#153] [mpv 通用]
发现了一个脚本，可以关闭时自动保存播放列表、播放状态，下次打开mvp时自动/手动加载，自己找了好久才找到。
https://github.com/CogentRedTester/mpv-scripts/blob/master/keep-session.lua

脚本放好位置后，可在mpv.conf加入启动自动恢复播放命令
script-opts-append=keep_session-auto_load=yes

或者在input.conf加入按键（r）手动恢复播放
##Keep Session脚本 恢复上次会话
 r                    script-message reload-session                #恢复上次会话（自动读取播放列表）

**Comment by dyphire** [comment]
对于这种需求我更推荐使用[simplehistory.lua](https://github.com/dyphire/Eisa01_mpv-scripts/blob/dev/scripts/simplehistory.lua)多功能历史播放记录脚本，可以绑定键位打开各种类型的历史记录播放列表。mpv空闲状态下按默认键位ctrl+r以恢复最后一次播放记录

同时可以搭配[history-bookmark.lua](https://github.com/dyphire/yuukidach-mpv-scripts/blob/master/history-bookmark.lua)脚本使用
该脚本功能：记录每个播放目录的最后一次播放文件，当使用mpv打开同目录其他文件时会在osd上提示是否恢复上次播放记录（所需键位需在input.conf中自行绑定）

以上脚本链接均为我个人修改版，上游pr尚未合并

**Reply by sqq135** [reply]
不好意思，问个比较简单的问题。
把lua和conf分别放入portable_config\script-opts和portable_config\scripts之后，还需要编辑哪些东西？
是只要编辑input.conf吗?还是说需要编辑simplehistory.conf？
能不能举个例子，比如input的例子。折腾半天没有效果，只能来请教，非常感谢

**Reply by LuvEven** [reply]
我也是刚开始用MPV，我直接把lua脚本放在portable_config\scripts文件夹下是生效的，我只放了lua没有添加放其他的conf配置文件

**Reply by sqq135** [reply]
是的是的，我之前就是纯小白，所以这个在mpvnet中没有效果我也不知道原因。就是你说的这个方法，非常感谢

**Reply by LuvEven** [reply]
大大我想问一下[history-bookmark.lua](https://github.com/dyphire/yuukidach-mpv-scripts/blob/master/history-bookmark.lua)中的435行
`msg = wait_msg .. " -- continue? " .. timeout .. " [EN/IG]"`
这个地方有实现什么功能吗，没看明白，如果是之前的enter进入n取消我就把这里注释掉了，每次重播都会显示

**Reply by dyphire** [reply]
无法复现，它不应该每次重播都提示，只有当同目录播放的文件不是上次最后播放的文件才会提示。这个消息的作用就是提示是否恢复目录上次播放，不能注释它否则这个脚本等于不存在

**Reply by LuvEven** [reply]
明白了，两个插件的作用我搞混了，绑定对应按键需要怎么绑定？

**Reply by dyphire** [reply]
参考 https://github.com/dyphire/mpv-config/blob/dcf1200bad117f5fec39aa5ff9f8456475c2ab5c/input.conf#L352-L354

**Reply by LuvEven** [reply]
非常感谢

**Reply by YUNDAN21** [reply]
> 对于这种需求我更推荐使用[simplehistory.lua](https://github.com/dyphire/Eisa01_mpv-scripts/blob/dev/scripts/simplehistory.lua)多功能历史播放记录脚本，可以绑定键位打开各种类型的历史记录播放列表。mpv空闲状态下按默认键位ctrl+r以恢复最后一次播放记录
> 
> 同时可以搭配[history-bookmark.lua](https://github.com/dyphire/yuukidach-mpv-scripts/blob/master/history-bookmark.lua)脚本使用 该脚本功能：记录每个播放目录的最后一次播放文件，当使用mpv打开同目录其他文件时会在osd上提示是否恢复上次播放记录（所需键位需在input.conf中自行绑定）
> 
> 以上脚本链接均为我个人修改版，上游pr尚未合并

大佬，我想请问一下，我在使用simplehistory.lua时，显示出的历史记录列表感觉有一种过曝的感觉，有点看不清，见下图，请问这个是因为什么呢？是需要调整什么地方的设置吗？
![image](https://github.com/user-attachments/assets/64a53148-1e7c-449a-a7da-db083001f90c)

**Reply by AkkarinME** [reply]
> 对于这种需求我更推荐使用[simplehistory.lua](https://github.com/dyphire/Eisa01_mpv-scripts/blob/dev/scripts/simplehistory.lua)多功能历史播放记录脚本，可以绑定键位打开各种类型的历史记录播放列表。mpv空闲状态下按默认键位ctrl+r以恢复最后一次播放记录
> 
> 同时可以搭配[history-bookmark.lua](https://github.com/dyphire/yuukidach-mpv-scripts/blob/master/history-bookmark.lua)脚本使用 该脚本功能：记录每个播放目录的最后一次播放文件，当使用mpv打开同目录其他文件时会在osd上提示是否恢复上次播放记录（所需键位需在input.conf中自行绑定）
> 
> 以上脚本链接均为我个人修改版，上游pr尚未合并

请问[simplehistory.lua](https://github.com/dyphire/Eisa01_mpv-scripts/blob/dev/scripts/simplehistory.lua)这个脚本怎么删除整个历史播放列表，我只能del键一条一条删

**Reply by sailovetheworld** [reply]
> 请问[simplehistory.lua](https://github.com/dyphire/Eisa01_mpv-scripts/blob/dev/scripts/simplehistory.lua)这个脚本怎么删除整个历史播放列表，我只能del键一条一条删

历史记录在\mpv-lazy\portable_config\mpvHistory.log这个文件里，可以直接编辑，不过我不确定是哪个插件生成的了
